core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -47,6 +48,8 @@ private[spark] class CoarseGrainedExecutorBackend(
   var executor: Executor = null
   @volatile var driver: Option[RpcEndpointRef] = None
 
+  private[this] val ser: SerializerInstance = env.closureSerializer.newInstance()
```

Comment:
should we make a thread local version of this? i'm slightly worried we might change this from ThreadSafeRpcEndpoint to a non-thread-safe version (i.e. multiple threads), and then forget to change this.